lib/lambda_ethereum_consensus/p2p/block_downloader.ex

```diff
@@ -13,29 +14,43 @@ defmodule LambdaEthereumConsensus.P2P.BlockDownloader do
 
   @impl true
   def init([host]) do
-    {:producer, host}
+    {:producer, {host, 0}}
   end
 
   @impl true
-  def handle_demand(incoming_demand, host) do
-    blocks =
-      for _ <- 0..incoming_demand do
-        request_block(host) |> wrap_message()
-      end
+  def handle_demand(incoming_demand, {host, remaining_demand}) do
+    demand = incoming_demand + remaining_demand
+    blocks = request_blocks(host, demand)
+    remaining = demand - length(blocks)
 
-    {:noreply, blocks, host}
+    if remaining > 0 do
```

Comment:
please comment why we might need to retry

Reply:
Done